client/src/components/Modal.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { ReactNode, useEffect, useRef } from "react";
+
+interface IModal {
+  closeModal: (value: boolean) => void;
+  children: ReactNode;
+}
+
+const Modal: React.FC<IModal> = ({ children, closeModal }) => {
+  return (
+    <div className="fixed top-0 left-0 right-0 w-screen h-screen z-50 overflow-x-hidden overflow-y-auto flex items-center justify-center bg-gray-700 bg-opacity-60">
+      <div className="bg-dark w-9/12 md:w-1/3 p-8 text-white rounded-2xl">
```

Comment:
i think to make it responsive just use max-w-[width value] and also w-full so that you don't have to resize it on every screen.. and add margin to the sides that's all. no media queries required.
eg.
className="bg-dark max-w-[30rem] w-full mx-4 p-8 text-white rounded-2xl"
the width value should come from the figma file, or you can pass it as a prop to make it dynamic for all modals